File: src/main/scala/com/sksamuel/scrimage/ImageWriter.scala
package com.sksamuel.scrimage

import java.io.{File, ByteArrayOutputStream, OutputStream}
import javax.imageio.ImageIO
import com.sksamuel.scrimage.Format.PNG
import org.apache.commons.io.{IOUtils, FileUtils}

/** @author Stephen Samuel */
class ImageWriter(out: OutputStream) {
    def write(image: Image, format: Format = PNG) {
        ImageIO.write(image.awt, "png", out)
        out.flush()
        IOUtils.closeQuietly(out)
    }
}

object ImageWriter {
    def apply(file: File) = new ImageWriter(FileUtils.openOutputStream(file))
    def apply(out: OutputStream) = new ImageWriter(out)
    def apply(bytes: Array[Byte]) = new ImageWriter(new ByteArrayOutputStream())
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/ImageWriter.java
package com.sksamuel.scrimage;

import java.io.IOException;
import java.io.OutputStream;

/**
 * Interface supporting writing of an Image to an array of bytes in a specified format, eg JPEG
 *
 * @see com.sksamuel.scrimage.nio.JpegWriter
 * @see com.sksamuel.scrimage.nio.PngWriter
 * @see com.sksamuel.scrimage.nio.GifWriter
 */
public interface ImageWriter {
   void write(AwtImage image, OutputStream out) throws IOException;
}
